src/jvmMain/kotlin/io/beatmaps/api/users.kt

```diff
@@ -919,21 +920,30 @@ fun Route.userRoute() {
             }
 
             transaction {
-                if (req.following) {
-                    Follows.upsert(conflictIndex = Follows.link) { follow ->
-                        follow[userId] = req.userId
-                        follow[followerId] = user.userId
-                        follow[since] = NowExpression(since.columnType)
-                        follow[upload] = req.upload
-                        follow[curation] = req.curation
-                    }
+                val shouldAlert = Follows.select { Follows.userId eq req.userId }.empty()
+
+                Follows.upsert(conflictIndex = Follows.link) { follow ->
+                    follow[userId] = req.userId
+                    follow[followerId] = user.userId
+                    follow[since] = NowExpression(since.columnType)
+                    follow[upload] = req.upload
+                    follow[curation] = req.curation
+                    follow[following] = req.following
+                }
+                val followedUser = if (shouldAlert) {
```

Comment:
This nullable into null coalescing seems a bit too esoteric to me.
Just wrap the whole thing in the condition...

```kt
if (shouldAlert) {
    val followedUser = UserDao.wrapRow(User.select { User.id eq req.userId }.single())

    if (followedUser.followAlerts) {
        Alert.insert(
            "New Follower",
            "@${user.uniqueName} is now following you!",
            EAlertType.Follow,
            req.userId
        )
    }
}
```

(single should also be safe enough as the insert should fail if the user id is invalid)